pytorch_lightning/plugins/training_type/sharded.py

```diff
@@ -41,7 +46,16 @@ def _reinit_optimizers_with_oss(self):
                 optimizer = optimizer._optimizer
             if not isinstance(optimizer, OSS):
                 optim_class = type(optimizer)
-                zero_optimizer = OSS(params=optimizer.param_groups, optim=optim_class, **optimizer.defaults)
+                is_fp16 = (self.lightning_module.trainer.accelerator_connector.precision == 16)
```

Comment:
the lightning module has the `precision` property which you can use: https://github.com/PyTorchLightning/pytorch-lightning/blob/80c529351439a0f8d3d6e9449cd47d16ba3abbec/pytorch_lightning/core/lightning.py#L96-L97

there's also the `precision` property on the trainer: https://github.com/PyTorchLightning/pytorch-lightning/blob/80c529351439a0f8d3d6e9449cd47d16ba3abbec/pytorch_lightning/trainer/properties.py#L396-L398

or `use_amp` on the trainer: https://github.com/PyTorchLightning/pytorch-lightning/blob/80c529351439a0f8d3d6e9449cd47d16ba3abbec/pytorch_lightning/trainer/properties.py#L140-L142

@justusschock @awaelchli which of these do you recommend here?

Reply:
I would stop at `trainer.precision` .